nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-value-providers/src/main/java/org/apache/nifi/stateless/parameter/aws/SecretsManagerParameterValueProvider.java

```diff
@@ -0,0 +1,172 @@
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * Reads secrets from AWS Secrets Manager to provide parameter values.  Secrets must be created similar to the following AWS cli command: <br/><br/>
+ * <code>aws secretsmanager create-secret --name "[ParamContextName]/[ParamName]" --secret-string '[ParamValue]'</code> <br/><br/>
+ *
+ * A standard configuration for this provider would be: <br/><br/>
+ *
+ * <code>
+ *      nifi.stateless.parameter.provider.AWSSecretsManager.name=AWS Secrets Manager Value Provider
+ *      nifi.stateless.parameter.provider.AWSSecretsManager.type=org.apache.nifi.stateless.parameter.aws.SecretsManagerParameterValueProvider
+ *      nifi.stateless.parameter.provider.AWSSecretsManager.properties.aws-credentials-file=./conf/bootstrap-aws.conf
+ * </code>
+ */
+public class SecretsManagerParameterValueProvider extends AbstractParameterValueProvider implements ParameterValueProvider {
+    private static final String QUALIFIED_SECRET_FORMAT = "%s/%s";
+    private static final String ACCESS_KEY_PROPS_NAME = "aws.access.key.id";
+    private static final String SECRET_KEY_PROPS_NAME = "aws.secret.access.key";
+    private static final String REGION_KEY_PROPS_NAME = "aws.region";
+
+    public static final PropertyDescriptor AWS_CREDENTIALS_FILE = new PropertyDescriptor.Builder()
+            .displayName("AWS Credentials File")
+            .name("aws-credentials-file")
+            .required(false)
+            .defaultValue("./conf/bootstrap-aws.conf")
+            .description("Location of the bootstrap-aws.conf file that configures the AWS credentials.  If not provided, the default AWS credentials will be used.")
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    private final Set<String> supportedParameterNames = new HashSet<>();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private AWSSecretsManager secretsManager;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    protected void init(final ParameterValueProviderInitializationContext context) {
+        super.init(context);
+
+        this.descriptors = Collections.singletonList(AWS_CREDENTIALS_FILE);
+
+        final String awsCredentialsFilename = context.getProperty(AWS_CREDENTIALS_FILE).getValue();
+        try {
+            this.secretsManager = this.configureClient(awsCredentialsFilename);
+        } catch (final IOException e) {
+            throw new IllegalStateException("Could not configure AWS Secrets Manager Client", e);
+        }
+
+        cacheSupportedParameterNames();
+    }
+
+    @Override
+    public boolean isParameterDefined(final String contextName, final String parameterName) {
+        return supportedParameterNames.contains(getSecretName(contextName, parameterName));
```

Comment:
Generally, the way that the parameters work in stateless, the context name should be considered optional. I.e., if the context name is "MyContext" and the parameter name is "MyParameter", we should return a value for "MyContext/MyParameter" if it exists, but if that doesn't exist, we should return the value for "MyParameter"

Reply:
Thanks for the additional context!  I'll make that update.